client/me/purchases/utils.js

```diff
@@ -9,8 +9,9 @@ import page from 'page';
 import analytics from 'lib/analytics';
 import paths from './paths';
 
+// TODO: Remove these property-masking functions in favor of selectors
 function getPurchase( props ) {
-	return props.selectedPurchase.data;
```

Comment:
Yeah, I added the `TODO` on the right for this, although you're right, we should just get rid of these altogether instead of using selectors. I think we should do it separately, as this PR is large.